Adapter issue 916, update installation docs (#603)
nunit/docs: docs/articles/vs-test-adapter/Adapter-Installation.md

# Adapter Installation

> The adapter is to be installed as a nuget package to each of your test projects.

Author: I don't see this text in the PR any longer?

# Adapter Installation

> The adapter is to be installed as a nuget package to each of your test projects.

Reviewer: ```suggestion
You'll need to install the adapter as a NuGet package for each of your test projects.
```

## Installing the NuGet Package

You will most likely want to add a new NUnit Test Project to your solution.  The easiest way is to use the ```dotnet``` command on the command line.

Go to your project root and where you want to add your project.

Create a folder for the project with a name matching what your new test project should be named.

Then run the command ````dotnet new nunit```

You will now get a complete nunit test project with the same name as the folder.  You will also have a template unit test class there as a starter.

> If you have a Visual Studio solution file in the root folder, you can go there and add the new nunit csproj easily:

Reviewer: ```suggestion
If you have a Visual Studio solution file in the root folder, you can go there and add the new nunit csproj:
```

Assume you are at a solution root, and you either have a solution file, or have just been adding it with ```dotnet new sln```

```cmd
md Whatever.Test
cd Whatever.Test

> dotnet new nunit

Reviewer: 💡 Should we add a note here about the fact that there are arguments and options to consider and running this command as-is may not produce the output folks are looking for, e.g. project name? Or should we maybe be more explicit with a comment that the project will be named after the directory?

Author: To me it can be either way, we're not going to do a complete course on dotnet, solutions and so, - on the other hand, it doesn't cost much to add what you suggest.  So, feel free :-)

cd ..
dotnet sln add Whatever.Test\Whatever.Test.csproj
```

And you're ready to go !

> ### Manually adding it as a package reference to your test projects

Reviewer: ```suggestion
### Manually Adding the Adapter as a Package Reference to Your Test Projects
```

> The recommended set of packages you should install is the framework, the adapter and the analyzer.  The adapter and the framework is required, the analyzer will help you, but is not required to make your project work.  The coverlet.collector is the recommended code coverage package you should go for.

Reviewer: ```suggestion
When adding packages manually, we recommended installing the framework, the adapter, and the analyzer for optimal out-of-the-box functionality.  The adapter and the framework are required. The analyzer will help during development, but is not strictly required to make your project work.  `coverlet.collector` is the recommended code coverage package you should go for.
```

> The recommended set of packages you should install is the framework, the adapter and the analyzer.  The adapter and the framework is required, the analyzer will help you, but is not required to make your project work.  The coverlet.collector is the recommended code coverage package you should go for.

Reviewer: ❓ Why is the coverlet package the recommended package?

Author: It is integrated into the Visual Studio Test platform as a collector, from VS version 16.5.

> You should ensure you have the ```Microsoft.NET.Test.Sdk``` in there too, as it is also required.

Reviewer: ```suggestion
You should ensure you also reference `Microsoft.NET.Test.Sdk` as well; it is required for test discoverability.
```

I wanted to add a little more detail; check me to make sure I'm right.

> 1. Open your test project csproj file

Reviewer: ```suggestion
1. Open your test project `csproj` file
```

> 2. Add the necessary package references as shown in the snippet example below (the versions should be the latest ones, the ones below is what is current at the date of writing)

Reviewer: ```suggestion
2. Add the necessary package references as shown in the snippet example below (NOTE: these are the current versions at time of writing; you'll want to install the latest versions.)
```

```xml
  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.11.0" />
    <PackageReference Include="NUnit" Version="3.13.2" />
    <PackageReference Include="NUnit3TestAdapter" version="4.1.0" />
    <PackageReference Include="coverlet.collector" Version="3.1.0" />
    <PackageReference Include="NUnit.Analyzers" Version="3.2.0" />
  </ItemGroup>
```

> Note: *You don't need to add any nunit.console or any other runner package*

Reviewer: We should probably change this to the note format seen elsewhere in the docs. Let me know if you need a reference to where we do it elsewhere 👍

### Working with the Visual Studio Nuget manager

> If you have a legacy project type, the easiest way is to use the Visual Studio 'Manage nuget packages in the solution' menu option. It can also be used for the new SDK projects, but it is actually faster to just open the csproj and copy paste the references in.

Reviewer: ```suggestion
If you have a legacy project type, or prefer working outside of the command line, you can also use the Visual Studio 'Manage NuGet packages in the solution' menu option. It can also be used for the new SDK projects, but you may find it faster to open the `csproj` and copy/paste the references in.
```

> You need to have an active solution, then follow these steps:

Reviewer: ```suggestion
With an active solution in Visual Studio, follow these steps:
```

> 1. From Tools menu, use Library Package Manager and select Manage NuGet packages for solution

Reviewer: ```suggestion
1. From the `Tools` menu, use Library Package Manager and select `Manage NuGet packages for solution`
```

> 2. In the left panel, select Online

Reviewer: ```suggestion
2. In the left panel, select `Online`
```

> 3. Locate (search for) NUnit3Test Adapter in the center panel and highlight it

Reviewer: ```suggestion
3. Locate (search for) `NUnit3Test Adapter` in the center panel and highlight it
```

> 4. Click 'Install'

Reviewer: ```suggestion
4. Click `Install`
```

> 5. In the "Select Projects" dialog, select all test projects in your solution.

Reviewer: ```suggestion
5. In the `Select Projects` dialog, select all test projects in your solution.
```

> 5. In the "Select Projects" dialog, select all test projects in your solution.

Reviewer: 💡 Should we add a final step of clicking OK and allowing the installation to complete? That might be helpful here.
